src/js/jsx/tools/SuperselectVectorMaskOverlay.jsx

```diff
@@ -0,0 +1,279 @@
+                        .attr("y", bounds.top + boundsAB.top)
+                        .attr("width", bounds.width)
+                        .attr("height", bounds.height)
+                        .attr("fill", "#3691ff")
+                        .attr("opacity", 0.5);
+                }
+            }
+        },
+
+        /**
+         * Draws the clickable icons on the vector Mask HUD
+         *
+         * @private
+         */
+        _drawVectorMaskHUDObjects: function () {
+            var flux = this.getFlux(),
+                uiStore = flux.store("ui"),
+                cloakRect = uiStore.getCloakRect();
+            
+            var layerTree = this.state.document.layers;
+    
+            var fluxActions = flux.actions,
+                toolStore = flux.store("tool"),
+                remToPx = flux.store("ui").remToPx,
+                numIcons = 3;
+
+            var sampleSize = remToPx(2.4),
+                iconSize = Math.round(sampleSize * 1),
+                iconOffset = Math.round(sampleSize / 1.5),
+                rectWidth = (sampleSize * numIcons) + (iconOffset * 5),
+                rectHeight = sampleSize + iconOffset * 2,
+                left = (cloakRect.left + cloakRect.right) / 2 - (rectWidth / 2),
+                top = (cloakRect.top + cloakRect.bottom) / 2 - (rectHeight / 2),
+                iconLeft = left + iconOffset,
+                iconTop = top + iconOffset,
+                rectRound = remToPx(1);
+
+            var rectTLX = Math.round(left),
+                rectTLY = Math.round(top);
+            
+            // Draw the frame
+            // A rounded rectangle
+            this._vectorMaskHudGroup
+                .append("rect")
+                .attr("x", rectTLX)
+                .attr("y", rectTLY)
+                .attr("width", rectWidth)
+                .attr("height", rectHeight)
+                .attr("rx", rectRound)
+                .attr("ry", rectRound)
+                .classed("vector-mask-hud", true)
+                .classed("vector-mask-hud-outline", true)
+                .on("click", function () {
+                    d3.event.stopPropagation();
+                });
+ 
+            // ellipse
+            // 
+            this._vectorMaskHudGroup
+                .append("use")
+                .attr("xlink:href", "img/ico-maskmode-ellipse.svg#maskmode-ellipse")
+                .attr("x", iconLeft)
+                .attr("y", iconTop)
+                .attr("width", iconSize)
+                .attr("height", iconSize)
+                .on("mouseover", function () {
+                    // Apply a temp mask on hover
+                    this.drawMaskPreview(this._vectorMaskHudGroup, layerTree, "ellipse");
+                    d3.event.stopPropagation();
+                }.bind(this))
+                .on("mouseout", function () {
+                    // Remove the temp mask on hover
+                    this.drawMaskPreview(this._vectorMaskHudGroup, layerTree, false);
```

Comment:
The JSDoc of this function states it accepts strings, if `false` is going to be a used value, please state that as well.